File: FinanceApi/Domain/Models/Fund.cs
using FinanceApi.Core.SpecialTypes;
using FinanceApi.Domain.Models.Base;

namespace FinanceApi.Domain.Models;

public class Fund : Entity<Guid>
{
    public Guid BankId { get; set; }

    public Guid CurrencyId { get; set; }

    public virtual Bank? Bank { get; set; }

    public virtual Currency? Currency { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime TimeStamp { get; set; }

    public Money Amount { get; set; } = 0m;

    public bool DailyUse { get; set; }
}

File: FinanceFrontEnd/FinanceApp/app/types/fund.ts
import type { BankData, CurrencyData } from './income';

export interface FundRecord {
  id: string;
  createdAt: string;
  updatedAt: string;
  timeStamp: string;
  deactivated: boolean;
  amount: number;
  bankId: string;
  currencyId: string;
  dailyUse: boolean;
  bank: BankData;
  currency: CurrencyData;
}
